game/src/main/kotlin/world/gregs/voidps/world/interact/entity/player/combat/consume/drink/Overload.kts

```diff
@@ -49,32 +72,37 @@ timerStart("overload") { player ->
 timerTick("overload") { player ->
     if (player.dec("overload_refreshes_remaining") <= 0) {
         cancel()
+        removeBoost(player)
```

Comment:
This should be in `timerStop`, in-case something else cancels the timer (command, death, etc..)

Reply:
Okay what I noticed what timer stop is called when you log out and your boosted stats aren't saved, do you think it'll be sufficient to removed boost on timer stop and apply boost on timer restart?

Reply:
Hmm it should save stats on logout, that might be a bug

Reply:
let me double check and i will update you here. just want to make sure i am right and i didn't do anything wrong.

Reply:
> Hmm it should save stats on logout, that might be a bug

Just double checked and the boosted is removed but the variable is stored in the player save

Reply:
Ah okay it's because timerStop is being called on logout, so it's removing the boost and the remaining refreshes variable, which is not correct.

```kotlin
timerStop("overload") { player ->
    if (logout) {
        return@timerStop
    }
    //...
}
```
Should fix it

Reply:
@GregHib i see you said " so it's removing the boost and the remaining refreshes variable" when the timer is stopped during logout it doesn't remove the remaining refreshes from the player, it still persists over logging. i have attached a gif, this might be a bug or i am misunderstanding what you're referring too.

![Peek 2024-05-13 17-16f](https://github.com/GregHib/void/assets/79747812/4d90d880-a903-4ccd-a6ae-726d2286745b)

Reply:
I was mainly referring to your levels which are being reset to 99, but it was also true for the refreshes remaining variable, but looks like you've removed that from the timerStop code, [it was there before](https://github.com/GregHib/void/blob/142a65daf4719cb1e8c1332303059e15eb460565/game/src/main/kotlin/world/gregs/voidps/world/interact/entity/player/combat/consume/drink/Overload.kts#L77).

It should still be there as a precaution